tools/analyse-storage/src/commands.rs

```diff
@@ -0,0 +1,228 @@
+
+static DEFAULT_HOME: Lazy<PathBuf> = Lazy::new(get_default_home);
+
+#[derive(Parser)]
+pub struct AnalyseStorageCommand {
+    #[clap(long, value_parser, default_value_os = DEFAULT_HOME.as_os_str())]
+    home: PathBuf,
+
+    #[arg(short, long)]
+    column: Option<String>,
+
+    #[arg(short, long)]
+    limit: Option<usize>,
+}
+
+pub fn get_db_col(col: &str) -> Option<DBCol> {
+    for db_col in DBCol::iter() {
+        if col_name(db_col) == col {
+            return Some(db_col);
+        }
+    }
+    return None;
+}
+
+fn get_all_col_family_names() -> Vec<DBCol> {
+    DBCol::iter().collect()
+}
+
+fn print_results(
+    sizes_count: &Vec<(usize, usize)>,
+    size_count_type: &str,
+    limit: usize,
+    total_num_of_pairs: usize,
+) {
+    println!(
+        "Total number of pairs read {}\n",
+        sizes_count.into_iter().map(|(_, count)| count).sum::<usize>()
+    );
+
+    // Print out distributions
+    println!("{} Size Distribution:", size_count_type);
+    println!(
+        "Minimum size {}: {:?}",
+        size_count_type,
+        sizes_count.iter().map(|(size, _)| size).min().unwrap()
+    );
+    println!(
+        "Maximum size {}: {:?}",
+        size_count_type,
+        sizes_count.iter().map(|(size, _)| size).max().unwrap()
+    );
+    println!("Most occurring size {}: {:?}", size_count_type, sizes_count.first().unwrap());
+    println!("Least occurring size {}: {:?}", size_count_type, sizes_count.last().unwrap());
+
+    let total_sizes_bytes_sum = sizes_count.iter().map(|a| a.0 * a.1).sum::<usize>();
+    println!(
+        "Average size {}: {:?}",
+        size_count_type,
+        total_sizes_bytes_sum as f64 / total_num_of_pairs as f64
+    );
+    let mut size_bytes_median = 0;
+    let mut median_index = total_num_of_pairs / 2;
+    for (size, count) in sizes_count.iter().take(limit) {
+        if median_index < *count {
+            size_bytes_median = *size;
+            break;
+        } else {
+            median_index -= count;
+        }
+    }
+    println!("Median size {} {}", size_count_type, size_bytes_median);
+    for (size, count) in sizes_count.iter().take(limit) {
+        println!("Size: {}, Count: {}", size, count);
+    }
+    println!("");
+}
+
+fn rocksdb_column_options(col: DBCol) -> Options {
+    let mut opts = Options::default();
+    opts.set_level_compaction_dynamic_level_bytes(true);
+
+    if col.is_rc() {
+        opts.set_merge_operator("refcount merge", RocksDB::refcount_merge, RocksDB::refcount_merge);
+        opts.set_compaction_filter("empty value filter", RocksDB::empty_value_compaction_filter);
+    }
+    opts
+}
+
+fn read_all_pairs(
+    db: &DB,
+    col_families: &Vec<String>,
+) -> (Vec<(usize, usize)>, Vec<(usize, usize)>) {
+    // Initialize counters
+    let key_sizes: Arc<Mutex<HashMap<usize, usize>>> = Arc::new(Mutex::new(HashMap::new()));
+    let value_sizes: Arc<Mutex<HashMap<usize, usize>>> = Arc::new(Mutex::new(HashMap::new()));
```

Comment:
Wrap into a single struct, also keep results separately - not printing immediately, and show it later on